Workflows: NonRetryableErrors Section Improvement (#17819)
cloudflare/cloudflare-docs: src/content/docs/workflows/build/sleeping-and-retrying.mdx

}, async () => { /* Step code goes here /* }
```

> ## Force a Workflow Instance to fail

Reviewer: ```suggestion
## Force a Workflow instance to fail
```

```

The Workflow instance itself will fail immediately, no further steps will be invoked, and the Workflow will not be retried.

> ## Avoid a Workflow instance to fail

Reviewer: ```suggestion
## Catch Workflow errors
```

> Any uncaught exceptions that propagate to the top level, or any steps that reach their retry limit, will cause the Workflow to end execution in an Errored state. However, this may not be intended, as for example, a cleanup of previous step executions may be required.

Reviewer: ```suggestion
Any uncaught exceptions that propagate to the top level, or any steps that reach their retry limit, will cause the Workflow to end execution in an `Errored` state.

If you want to avoid this, you can catch exceptions emitted by a `step`. This can be useful if you need to trigger clean-up tasks or have conditional logic that triggers additional steps.
```

> To allow the Workflow to continue its execution, surround the intended steps that are allowed to fail with a try/catch.

Reviewer: ```suggestion
To allow the Workflow to continue its execution, surround the intended steps that are allowed to fail with a `try-catch` block.
```

> Simple example of a Worflow with a cleanup step:

Reviewer: ```suggestion
```

```ts
...
await step.do('task', async () => {
	// work to be done
});

try {
    await step.do('non-retryable-task', async () => {
		// work not to be retried
        throw new NonRetryableError('oh no');
    });

> } catch {

Reviewer: ```suggestion
} catch(e as Error) {
```

>     console.log('error was thrown');

Reviewer: ```suggestion
    console.log(`Step failed: ${e.message}`);
```

}

// the Workflow will not fail and will continue its execution

> await step.do('cleanup-task', async() => {

Reviewer: I would put this into the `catch` block?

```

The Workflow instance itself will fail immediately, no further steps will be invoked, and the Workflow will not be retried.

> Users can catch this error if they intend to continue the Workflow execution. Otherwise, the Workflow will fail.

Reviewer: ```suggestion
Catching this exception within the same step as it was thrown will cause it to be ignored. If the exception propagates outside of the step, the step and the Workflow instance itself will subsequently fail.
```

> Users can catch this error if they intend to continue the Workflow execution. Otherwise, the Workflow will fail.

Reviewer 2: ```suggestion
Like other Errors, any uncaught exceptions that propagate to the top level will cause the Workflow to end execution in an Errored state. Catch the `NonRetryableError` to continue Workflow execution. 
```
